commit/merkleroot/observation.go

```diff
@@ -474,28 +508,40 @@ func (o observerImpl) ObserveMerkleRoots(
 	return roots
 }
 
-// computeMerkleRoot computes the merkle root of a list of messages
+// computeMerkleRoot computes the merkle root of a list of messages.
+// Messages should be sorted by sequence number and not have any gaps.
 func (o observerImpl) computeMerkleRoot(ctx context.Context, msgs []cciptypes.Message) (cciptypes.Bytes32, error) {
-	var hashes [][32]byte
-	sort.Slice(msgs, func(i, j int) bool { return msgs[i].Header.SequenceNumber < msgs[j].Header.SequenceNumber })
+	hashes := make([][32]byte, len(msgs))
+	hashesStr := make([]string, len(hashes)) // also keep hashes as strings for logging purposes
 
+	eg := &errgroup.Group{}
```

Comment:
Does this really need parallelization?

Reply:
Eh I think the benchmarks would be misleading anyway, the benchmark has the whole PC to run, in prod there's way more CPU overhead / thread spawning / goroutine ctx switching. I'm fine with it though, doesn't seem to hurt that much.